src/com/twitter/intellij/pants/service/project/PantsResolver.java

```diff
@@ -93,6 +100,30 @@ public void resolve(
     }
   }
 
+  public void resolvePythonEnvironment(
+    ProjectData projectData,
+    List<String> selectedTargets,
+    ProcessAdapter processAdapter
+  ) {
+    PythonSetup pythonSetup = myProjectInfo.getPythonSetup();
+    boolean needsPython = myProjectInfo.getTargets().values().stream().anyMatch(t -> t.isPythonTarget());
+    if (needsPython && pythonSetup != null) {
+      PythonVenvBuilder.forProjectPath(myExecutor.getBuildRoot().toString(), processAdapter).ifPresent(builder -> {
+        String target = mainTargetName(selectedTargets);
+        Path venvDir = Paths.get(projectData.getIdeProjectFileDirectoryPath(), "venv");
+        PythonInterpreterInfo python = builder.build(target, venvDir);
+        String environmentName = "Python virtual environment from venv_builder";
+        pythonSetup.getInterpreters().put(environmentName, python);
+        pythonSetup.setDefaultInterpreter(environmentName);
+      });
+    }
+  }
+
+  private String mainTargetName(List<String> selectedTargets){
+    //FIXME can there be more than one target?
```

Comment:
+1

Reply:
The venv_builder tool supports only one target selector (which, however, might be as general as `::`), so I think this way of selecting the "main" target is an acceptable compromise for now.